src/payment-request.ts

```diff
@@ -54,27 +79,69 @@ export async function createPaymentRequest(
           }
         }
       `,
-      { sessionId, payeeName, payeeEmail, sessionPreviewData }
+      {
+        sessionId,
+        payeeName,
+        payeeEmail,
+        sessionPreviewData: validatedSessionData.sessionPreviewData,
+        //TODO amount: validatedSessionData.amount
+      }
     );
   return response?.insert_payment_requests_one;
 }
 
-export function buildSessionPreviewData(
-  session: Session,
-  sessionPreviewKeys: KeyPath[]
-): PaymentRequest["sessionPreviewData"] {
+// lookup the "amount" passport key from the pay component of a given flow
+async function getPaymentAmountKey(
+  client: GraphQLClient,
+  flowId: string
+): Promise<string> {
+  const flowGraph = await getLatestFlowGraph(client, flowId);
+  if (!flowGraph) {
+    throw new Error("flow not found");
+  }
+  // search for pay component from the bottom of the root node upwards
+  let payNode: Node | undefined;
+  flowGraph._root.edges.reverse().forEach((rootNodeId) => {
+    const node = flowGraph[rootNodeId];
+    if (node.type === ComponentType.Pay) {
+      payNode = node;
+      return;
+    }
```

Comment:
on first read, it looks like this is only checking node types that are directly on the "root" and would fail to find a Pay component on a branch, is that right? 

LDC content looks like this & will need to account for branches! But the bottom-up search direction is clever here 
![Screenshot from 2023-04-06 10-18-51](https://user-images.githubusercontent.com/5132349/230318246-0611a585-91f8-4afc-8a96-9ab7e66ec93c.png)

Reply:
This sounds like a good plan :heavy_check_mark:  

And for clarity: Sections are the only node type that we currently validate in a graph, all other node types including Pay can exist in any position in a graph and have zero or many instances. So searching breadcrumb rather than flow is definitely a good shout and the more efficient way forward based on our graph assumptions :+1:

Reply:
Update from dev call - we'll add a breadcrumb at the API level to achieve this.

Reply:
I found a slightly different strategy for this PR (see below). If it's still helpful to modify breadcrumbs, that's doable too.